corehq/apps/locations/permissions.py

```diff
@@ -321,3 +321,18 @@ def _inner(request, domain, loc_id, *args, **kwargs):
         return location_restricted_response(request)
 
     return require_can_edit_or_view_locations(_inner)
+
+
+def can_access_location_data(view_fn):
+    """
+    Decorator controlling a user's access to VIEW data for a specific location.
+    """
+    @wraps(view_fn)
+    def _inner(request, domain, *args, **kwargs):
+        loc_id = request.GET.get('location_id')
```

Comment:
what happens if `location_id` isn't in the URL?

Reply:
it will skip the check

Reply:
That would allow people to easily circumvent this access control by modifying the request parameters.  The failure mode here should be to default to no access - that's how the other utilities here operate.

More broadly, this is half of an interface - it only works with views which accept `location_id` as a URL parameter.  It's essentially the same as `can_edit_location`, except that the location ID is passed as a keyword arg rather than as a component of the URL.  I'd advocate for either using the same decorator and convention in both instances or moving this utility to `icds_reports`.  Otherwise there's ambiguity in which is the "correct" way to do this.

Reply:
hey, @esoergel , for my knowledge, does not `can_edit_location` checks and restricts user to have edit access. Asking because of most of the Dashboard users might not have the edit access to edit the locations. But still can view the dashboard. do you think `can_edit_or_view_location` is better alternative?

Reply:
Good point - it sounds like `can_edit_or_view_location` would be more appropriate in this case.

Reply:
@challabeehyv  could you please reuse can_edit_or_view_location decorator is possible. Else, if possible to reuse it because it needs loc_id as an argument and your decorator does not.